### ISSUE TITLE
[Bug]: Wrong URL on BucketApi.downloadPublic()

### ISSUE THREAD

**Issue by denchic45** [supabase-community/supabase-kt#47]
### General Info

- [X] I installed the latest version of Supabase-Kt
- [X] I checked for similar bug report

### What happened? (include your code)

A try to download file by url:
https://twmjqqkhwizjfmbebbxj.supabase.co/storage/v1/object/public/main/courses/0e324582-b99b-4be2-a48d-c651b45ec797/elements/6c868ee2-f13d-499b-8a61-c7f5b4c03221/data.txt
But a get error because the request is built incorrectly:
https://twmjqqkhwizjfmbebbxj.supabase.co/storage/v1/https://twmjqqkhwizjfmbebbxj.supabase.co/storage/v1/object/public/main/courses/0e324582-b99b-4be2-a48d-c651b45ec797/elements/6c868ee2-f13d-499b-8a61-c7f5b4c03221/data.txt

I haven't checked, but I'm guessing there's also a problem with the request url build in the 'downloadAuthenticated' method:

### Platform(s)

Desktop

**Comment by jan-tennert** [comment]
That is weird, do other methods work?

**Comment by jan-tennert** [comment]
How do you even download it? You put the path in the download method, not the whole url.

**Comment by jan-tennert** [comment]
So I guess for you it would be
```kotlin
api.downloadAuthenticated("main/courses/0e324582-b99b-4be2-a48d-c651b45ec797/elements/6c868ee2-f13d-499b-8a61-c7f5b4c03221/data.txt")
```
Same for downloadPublic

**Comment by denchic45** [comment]
Yes, everything is so. I only pass the path
```
bucket.downloadPublic("courses/920a4d5b-2e20-4ba7-986f-3af522e37421/elements/2c78d4c6-0076-4c48-8cfc-c813c14e567b/data.txt ")
```

I checked it through the debugger, at the moment in this function the url is still correct, but then apparently the base url is added again
![image](https://user-images.githubusercontent.com/80104523/214026184-f8925c03-8869-4e53-ba23-e5cc280408c6.png)

**Comment by denchic45** [comment]
I'm guessing the error is that the 'publicUrl(path: String)' function should just return `"object/public/$bucketId/$path"`, without the base url, which will be attached later.

As a result in 'storage.api.get(url).body()'
`url `should actually be`"object/public/$bucketId/$path"`

**Comment by jan-tennert** [comment]
Nvm found the problem, will publish a fix later

**Comment by jan-tennert** [comment]
Should be fixed in 0.7.6, let me know if there are any issues!
